### ISSUE TITLE
URL starting with {param}-

### ISSUE THREAD

**Issue by theo744** [airbnb/DeepLinkDispatch#257]
Hi,
I have an issue with an url starting with {param}-

For example : https://domain.app/7-something
I can't catch it with @DeepLink("https://domain.app/{id}-something")

If I change the URL to:
https://domain.app/7/something
I can catch it with @DeepLink("https://domain.app/{id}/something")

Why is that ?

Theo

**Comment by MichaelEvans** [comment]
Would it be possible to make the entire argument the `id`? For example: 
```
@Deeplink("https://domain.app/{id}")
```

And then parse the `id` for the part you're interested in before the `-something`?

**Comment by theo744** [comment]
Yes, it would probably work. The problem is that I already have other links handled successfully with @DeepLink, so I would have to parse all my other links with the global {id}, that's quiet a lot of work...

Do you know what is the matter with {id}-something ? It seems related to the '-' character but I don't know...

Théo
